### PR TITLE
Add PHP 7.3 preview (xdebug in rc version)

### PR DESCRIPTION
**Pull request by andrerom**
Calling this preview as memcached is held up from being released on PHP 7.3 atm, see inline link.

### CODE REVIEW

**Comment by andrerom** [comment]
First failure: difference in libzip usage during build, could have given it options to use bundled version but [UPGRADE](https://github.com/php/php-src/blob/PHP-7.3/UPGRADING) suggests not doing that

Second failure: Blackfire not found for 7.3, 1h after the build [they tweeted support was added](https://twitter.com/blackfireio/status/1074607619346759680).

Third failure: xdebug, trying beta channel even if that is from september, but results in app_1  | `Segmentation fault (core dumped)`

Summary:
- Need to wait for stable xdebug
- which waits for Travis-ci support for PHP 7.3 on Xenial flavour: https://travis-ci.community/t/please-add-php-7-3-on-xenial/1238/14
- which waits for php-memcached for 7.3: https://github.com/php-memcached-dev/php-memcached/issues/408
- which waits for php-msgpack for PHP 7.3: [2.0.3 just out](https://pecl.php.net/package-info.php?package=msgpack&version=2.0.3) + time to release

**Comment by sodabrew** [comment]
memcached 3.1.0 is released!
https://pecl.php.net/package-changelog.php?package=memcached&release=3.1.0

File: bin/.travis/test.sh [hunk 1]
```diff
@@ -78,18 +78,9 @@ docker run -ti --rm \
...
 printf "\Integration: Behat testing on ez_php:latest and ez_php:latest-dev with eZ Platform\n"
 cd volumes/ezplatform
 
-# Tag image alias for what (exactly) eZ Platform is currently using in order to be able to test across branches
-## As we don't want it to pull in remote, but rather use what we just built here
-## NOTE: On larger changes to this images, we would need to pull in custom branches with adaptions on Platform side as well as below
-## TODO: The tag aliases here will not be needed once eZ Platform uses PHP_IMAGE variable within Dockerfile's from section
-docker tag ez_php:latest "ezsystems/php:7.2-v1"
-docker tag ez_php:latest "ezsystems/php:7.1-v1"
-docker tag ez_php:latest "ezsystems/php:7.0-v1"
```
Comment (andrerom) [inline bin/.travis/test.sh:91]: FYI: This is removed as it's not needed anymore as the configurable injection of this is now part of both 1.13 and 2.x releases:
https://github.com/ezsystems/ezplatform/commit/cb6794856afa0e00a3ad727e0b71e3b0b865a786

**Comment by andrerom** [comment]
@mnocon @vidarl I think this is ready for review, even if xdebug is still unstable (rc) I think we can go ahead here to start to test out PHP 7.3 in other repos.

**Review by mnocon — APPROVED** [review]
Looks good to me.

One question that came to my mind, not really related to this PR:
To start using Node on Travis, I've simply set `PHP_IMAGE` to the Node flavour (https://github.com/ezsystems/ezplatform/pull/345/files#diff-f579cccc964135c7d644c7b2d3b0d3ecR13).

Does it mean, that for v2 testing this line https://github.com/ezsystems/docker-php/pull/32/files#diff-6757d9ef6e7f3b3c85ea7eb11b9af47fR84 should be changed and PHP_IMAGE should be set based on EZ_VERSION (ez_php:latest for v1 and ez_php:latest-node for testing on version 2.5)? If yes I can create a follow-up for that.

**Comment by sodabrew** [comment]
Is the text specifying `w/o memcached` still accurate? I see the memcached extension is built in this Dockerfile.

**Comment by andrerom** [comment]
@sodabrew Github usability bug, I renamed the PR title to update state, then hit merge and the title of the merge commit reflect the old title of the PR. Reported it to Github several months ago ;)

**Comment by sodabrew** [comment]
The README still says no memcached at https://github.com/ezsystems/docker-php#images.

**Comment by andrerom** [comment]
> The README still says no memcached however.

Good catch: https://github.com/ezsystems/docker-php/commit/e7534b3e14445625c44c3c25215cf8165dd3caa5

> One question that came to my mind

@mnocon hmm,
what I had in mind was that we use node image in these two places:
- During build phase of Docker-app, but *not* in final image
    - So here: https://github.com/ezsystems/ezplatform/blob/master/doc/docker/Dockerfile-app#L2
   - But *not* here: https://github.com/ezsystems/ezplatform/blob/master/doc/docker/Dockerfile-app#L31
- For dev image: https://github.com/ezsystems/ezplatform/blob/master/doc/docker/base-dev.yml#L6

So just fixing the env variables won't do.
I didn't check, but does Docker in these places allow us to concat? So we have a PHP_BASE_IMAGE env/arg that we add "-node" or "-dev" to when we need a specific flavour (?)

**Comment by mnocon** [comment]
Yes, it is possible. Something like:
``` 
(base-prod.yml)
version: '3.3'
# Simple single server setup for prod

services:
    app:
      build:
        context: .
        dockerfile: ${APP_DOCKER_FILE}
        args:
         - PHP_IMAGE
```
``` 
(.env)
PHP_IMAGE=ezsystems/php:7.2-v1
APP_DOCKER_FILE=Dockerfile-app
```
```
(Dockerfile-app)
ARG PHP_IMAGE=ezsystems/php:7.2-v1-dev
FROM ${PHP_IMAGE}-node

RUN php -v
RUN node -v
```

builds the image with Node flavour.

I'll amend my PR with that and ping you there, thanks!

**Pull request merged**